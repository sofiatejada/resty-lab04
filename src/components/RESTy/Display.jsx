import React from 'react';
import PropTypes from 'prop-types';
import ReactJson from 'react-json-view';

export default function Display(response) {
  return (
    <div aria-label="res" role="stuff">
      <ReactJson src={response} />
    </div>
  );
}

Display.propTypes = {
  display: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};
