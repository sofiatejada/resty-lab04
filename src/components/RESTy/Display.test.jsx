import React from 'react';
import { render, screen } from '@testing-library/react';
import Display from './Display';

describe('display', () => {
  it.skip('renders display', () => {
    const hola = {
      'Hola': 'Rock, Paper, REST!'
    };

    render(
      <Display display={hola} />
    );
    
    const display = screen.getByText('"Rock, Paper, REST!"');
    expect(display).toMatchSnapshot();
  });
});
