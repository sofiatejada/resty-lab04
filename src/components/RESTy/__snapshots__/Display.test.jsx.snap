// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display renders display 1`] = `
<span
  class="string-value"
  style="cursor: default;"
>
  "
  Rock, Paper, REST!
  "
</span>
`;
